### PR TITLE
HDDS-6154. Update NodeManagerInsight with the correct node metrics

### PR DESCRIPTION
**Pull request by VarshaRaviCV**
## What changes were proposed in this pull request?

Updating the NodeManagerInsight to reflect the correct node metrics

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-6154

## How was this patch tested?

Docker-compose environment with command ozone insight metrics scm.node-manager
`sh-4.2$ ozone insight metrics scm.node-manager
Metrics for `scm.node-manager` (SCM Datanode management related information.)

Node counters

  in_service healthy Nodes: 3
  in_service stale Nodes: 0
  in_service dead Nodes: 0
  decommissioning healthy Nodes: 0
  decommissioning stale Nodes: 0
  decommissioning dead Nodes: 0
  decommissioned healthy Nodes: 0
  decommissioned stale Nodes: 0
  decommissioned dead Nodes: 0
  entering_maintenance healthy Nodes: 0
  entering_maintenance stale Nodes: 0
  entering_maintenance dead Nodes: 0
  in_maintenance healthy Nodes: 0
  in_maintenance stale Nodes: 0
  in_maintenance dead Nodes: 0

HB processing stats

  HB processed: 26
  HB processing failed: 0`

### CODE REVIEW

**Review by adoroszlai — COMMENTED** [review]
Thanks @VarshaRaviCV for working on this.

I would like to suggest creating these `MetricDisplay` instances in a nested loop (over operational state and health state) instead of manually enumerating all of them.  This will reduce future work required for adapting to related changes.  It will also take care of ordering these metrics logically (e.g. "stale" should come before "dead", "entering maint" should come before "in maint").

File: hadoop-ozone/insight/src/main/java/org/apache/hadoop/ozone/insight/scm/NodeManagerInsight.java [hunk 1]
```diff
@@ -50,39 +51,19 @@ public List<MetricGroupDisplay> getMetrics(Map<String, String> filters) {
...
+
+    for(int i=0; i < operationalState.length; i++){
+      for(int j=0; j < healthState.length; j++){
+        String metricId = String.format("scm_node_manager_%s_%s_nodes",
+            operationalState[i], healthState[j]);
+        String metricDescription = String.format("%s %s Nodes",
+            operationalState[i], healthState[j]);
+        nodes.addMetrics(new MetricDisplay(metricDescription, metricId));
+      }
+    }
```
Comment (adoroszlai) [inline hadoop-ozone/insight/src/main/java/org/apache/hadoop/ozone/insight/scm/NodeManagerInsight.java:66]: We can use enums directly:

```suggestion
    for (NodeOperationalState opState : NodeOperationalState.values()) {
      for (NodeState healthState : NodeState.values()) {
        String metricId = String.format("scm_node_manager_%s_%s_nodes",
            opState, healthState).toLowerCase();
        String metricDescription = String.format("%s %s Nodes",
            opState, healthState);
        nodes.addMetrics(new MetricDisplay(metricDescription, metricId));
      }
    }
```

**Review by adoroszlai — COMMENTED** [review]
Thanks @VarshaRaviCV for updating the patch.

File: hadoop-ozone/insight/src/main/java/org/apache/hadoop/ozone/insight/scm/NodeManagerInsight.java (hunk 1 above)
Comment (VarshaRaviCV) [reply hadoop-ozone/insight/src/main/java/org/apache/hadoop/ozone/insight/scm/NodeManagerInsight.java]: Thanks for the suggestion @adoroszlai . I have made the changes.

**Comment by adoroszlai** [comment]
Thanks @VarshaRaviCV for updating the patch.  @sodonnel please review.

**Pull request merged**